<?xml version="1.0" encoding="UTF-8"?>
<!--
  Created by Ian McCubbin, 10/19/2024
  Main window for Space Invaders Emulator, contains options to play,
  set controls, and show instructions

  Modified by Ian McCubbin, 10/23/2024
  - Integrated instructions page
  - Additional optimizations to main window for input handling

  Modified by Ian McCubbin, 10/23/2024
  - Integrated settings page

  Modified by Ian McCubbin, 10/28/2024
  - Added ability to read keymap.json to set game key controls.
-->

<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>845</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Space Invaders</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-image: url(:/Images/images/spcaeSky.jpg);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayoutMain">
    <item>
     <widget class="QFrame" name="frame">
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgba(255, 255, 255, 0);</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Sunken</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayoutButtons">
       <item>
        <widget class="QPushButton" name="buttonPlay">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Magneto</family>
           <pointsize>28</pointsize>
           <italic>false</italic>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: transparent;
color: rgb(170, 255, 0);
border: 4px solid rgb(170,255,0)
          </string>
         </property>
         <property name="text">
          <string>Play Game</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonSettings">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Magneto</family>
           <pointsize>28</pointsize>
           <italic>false</italic>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: transparent;
color: rgb(170, 255, 0);
border: 4px solid rgb(170,255,0)
          </string>
         </property>
         <property name="text">
          <string>Settings</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonInstructions">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Magneto</family>
           <pointsize>28</pointsize>
           <italic>false</italic>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: transparent;
color: rgb(170, 255, 0);
border: 4px solid rgb(170,255,0)
          </string>
         </property>
         <property name="text">
          <string>Instructions</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
